### PR TITLE
notebook 01

### PR DESCRIPTION
**Pull request by clonker**

### CODE REVIEW

File: notebooks/06-hmm.ipynb [hunk 1]
```diff
@@ -75,6 +77,23 @@
     "We include an implied timescales plot."
    ]
   },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def its_bmsm(dtrajs, lags, nits=None):\n",
```
Comment (thempel) [inline notebooks/06-hmm.ipynb:86]: 4th

File: notebooks/06-hmm.ipynb [hunk 2]
```diff
@@ -163,8 +187,10 @@
     "axes[0].set_ylabel('$y$')\n",
     "axes[0].set_title('discretization')\n",
     "\n",
-    "pyemma.plots.plot_implied_timescales(\n",
-    "    pyemma.msm.its(poor_dtraj, lags=lags, errors='bayes'), ylog=False, ax=axes[1])\n",
+    "lags = [i for i in range(1, 10)]\n",
```
Comment (thempel) [inline notebooks/06-hmm.ipynb:190]: `np.arange(1, 10)`?

File: notebooks/06-hmm.ipynb [hunk 3]
```diff
@@ -190,8 +216,26 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "its_hmm_poor = pyemma.msm.timescales_hmsm(poor_dtraj, nstates=2, lags=lags, errors='bayes')\n",
-    "its_hmm_good = pyemma.msm.timescales_hmsm(good_dtraj, nstates=2, lags=lags, errors='bayes')"
+    "def its_hmm(dtraj, lags, nits=None, n_hidden_states=2):\n",
```
Comment (thempel) [inline notebooks/06-hmm.ipynb:219]: does this only work with one dtraj? (5th)

File: notebooks/06-hmm.ipynb [hunk 4]
```diff
@@ -247,12 +287,17 @@
    },
    "outputs": [],
    "source": [
-    "poor_hmm = pyemma.msm.estimate_hidden_markov_model(poor_dtraj, 2, lag=1)\n",
-    "good_hmm = pyemma.msm.estimate_hidden_markov_model(good_dtraj, 2, lag=1)\n",
+    "init_poor = dt.markov.hmm.init.discrete.metastable_from_data(poor_dtraj, n_hidden_states=2, lagtime=1)\n",
```
Comment (thempel) [inline notebooks/06-hmm.ipynb:290]: We need to add some text here, that's a lot done in one cell (already done in ITS function above). I guess it requires to add some more info about the algorithmic details so people understand what it does.

**Review by thempel — COMMENTED** [review]
A great step ahead. My major points are:
- There are 5 different implied timescales functions that are partially redundant. I think we should either add such a function to deeptime or write 2 consistent functions for (B)MSM and (B)HMM here in the notebooks.
- From my experience, it is super important to keep concatenated trajectories out of the game as far as possible because students have a very strong tendency to use them for estimating MSMs.

File: notebooks/01-io-features-presentation.ipynb [hunk 5]
```diff
@@ -243,14 +244,15 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "tica = pyemma.coordinates.tica(Y, lag=10, var_cutoff=0.95)"
+    "tica_estimator = dt.decomposition.TICA(lagtime=10, var_cutoff=0.95)\n",
```
[the remark this answers is not in the document]
Comment (thempel) [reply notebooks/01-io-features-presentation.ipynb:247]: Okay, `tica` var name is consistent with the rest, so let's keep it.

File: notebooks/01-io-features-presentation.ipynb [hunk 6]
```diff
@@ -383,10 +391,8 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "clustering = pyemma.coordinates.cluster_kmeans(tics, \n",
-    "                                               k=75,\n",
-    "                                               stride=50,  # very large stride for presentation purposes only\n",
-    "                                               max_iter=30)"
+    "kmeans_estimator = dt.clustering.Kmeans(n_clusters=75, max_iter=30)\n",
+    "clustering = kmeans = kmeans_estimator.fit(np.concatenate(tics)).fetch_model()"
```
[the remark this answers is not in the document]
Comment (clonker) [reply notebooks/01-io-features-presentation.ipynb:395]: oopise

File: notebooks/02-io-features-hands-on.ipynb [hunk 7]
```diff
@@ -397,16 +380,9 @@
...
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "metadata": {},
-   "outputs": [],
-   "source": [
-    "pca.dimension()"
+    "from sklearn.decomposition import PCA\n",
```
[the remark this answers is not in the document]
Comment (clonker) [reply notebooks/02-io-features-hands-on.ipynb:383]: i dont mind either way, just thought the estimator api of deeptime was closer to sklearn than pyemma

File: notebooks/01-io-features-presentation.ipynb [hunk 8]
```diff
@@ -321,11 +323,12 @@
    ]
   },
   {
-   "cell_type": "code",
-   "execution_count": null,
+   "cell_type": "markdown",
    "metadata": {},
-   "outputs": [],
    "source": [
+    "<span style=\"color:red\">*we could also leave it like this instead of using the built-in method*</span>\n",
```
[the remark this answers is not in the document]
Comment (clonker) [reply notebooks/01-io-features-presentation.ipynb:329]: cool ill leave it with the built-in then

**Pull request merged**